### ISSUE TITLE
[SCM-531] Incorrect SCM URL constructed when using parent POM that is not in the parent directory

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#744]
**[Darren Bell](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=kango_v)** opened **[SCM-531](https://issues.apache.org/jira/browse/SCM-531?redirect=false)** and commented

In our parent pom we have this:

```
<properties>
  <scm.module>master-pom</scm.module>
</properties>
<scm>
  <connection>scm:cvs:ext:build@server:/srv/cvs/:${scm.module}</connection>
  <developerConnection>scm:cvs:ext:build@server:/srv/cvs/:${scm.module}</developerConnection>
  <url>http://server/fe/browse/UK-CVS-Rsync/${scm.module}</url>
</scm>
```

Our child pom (not in the same directory), has this:

```
<artifactId>dd-util</artifactId>
<properties>
  <scm.module>util</scm.module>
</properties>
```

Of course, the child pom is overriding the property.  Well, it sort of works.
When building the site of the parent pom, we get:

```
http://server/fe/browse/UK-CVS-Rsync/master-pom
```

... which is correct.  But when building the site for the child pom, we get:

```
http://server/fe/browse/UK-CVS-Rsync/util/dd-util
```

I'm quite confused as to why it whould be appending the artifactId of the child project.  The property is getting replaced ok, but the the extra bit?  This has got to be a bug.

I'm not releasing the project, just building the site.

Any ideas?  If this is against the wrong project, can it be moved?

---

17 votes, 15 watchers

**Comment by jira-importer** [comment]
**[João Rafael Nicola](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=joaoraf)** commented

I am also experiencing a similar effect.

My parent pom has, for example:

\<developerConnection>scm:svn:https://svnserver/myprojects/${project.artifactId}/trunk\</developerConnection>

When running "mvn help:effective-pom" on the child, whose artifactId is, for example, child-project, I get:

\<developerConnection>scm:svn:https://svnserver/myprojects/child-project/trunk/child-project\</developerConnection>

instead of the expected:

\<developerConnection>scm:svn:https://svnserver/myprojects/child-project/trunk\</developerConnection>

For some reason, the developerConnection property gets appended with the artifactId when inherited. Although this makes sense when dealing with a multi-module project which is versioned in a single project structure in subversion, it does not when the parent pom is used just to standardize some properties on the pom, which each child project independently versioned.

I think there should be some way to disable this behaviour.

João Nicola

**Comment by jira-importer** [comment]
**[Tomas Varaneckas](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tomas.varaneckas)** commented

An ugly workaround for that is to have links ending with # or ?foo=, so finally they work even with /artifact-id in the end. I.e.:

    <scm>
        <connection>scm:git:git://git.xxx.com/${xxx.project.path}.git?proj=</connection>
        <developerConnection>scm:git:git://gitosis@git.xxx.com/${xxx.project.path}.git?proj=</developerConnection>
        <url>http://git.xxx.com/?p=${xxx.project.path}.git;a=summary#</url>
    </scm>

**Comment by jira-importer** [comment]
**[Youri Bonnaffé](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=youribonnaffe)** commented

I experienced the same behavior as Joao.

**Comment by jira-importer** [comment]
**[Adrian Moerchen](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=adrian2k7)** commented

This is still an issue in Maven 3.0.5

**Comment by jira-importer** [comment]
**[Laurent TOURREAU](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=lautou)** commented

This issue has also been detected in Maven 3.1.1

**Comment by jira-importer** [comment]
**[Rami Ojares](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rojares)** commented

Still there in version 3.3.3
The point is that inheritance does not work here correctly when using properties.
Try putting \<scm>\<developerConnection>${project.artifactId}\</developerConnection>\</scm>
in parent pom and see what the effective pom says in the child project.
Shows something wrong, that's for sure.

Is anybody listening?
This was reported 2010 but is still not assigned.

**Comment by jira-importer** [comment]
**[Eduardo Hernández Guerra](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=eduardomhg)** commented

I would also like this to be fixed or at least any workaround (the ugly one proposed above by Tomas does not work with SVN).

Any way of disabling the automatic appending of the artifactId to the inherited scm connection strings?

**Comment by jira-importer** [comment]
**[jycr](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jcronier)** commented

It seems that it "works as designed" :
https://stackoverflow.com/questions/20513744/project-artifactid-in-parent-pom-xml-resolves-odd#answer-20517630

**Comment by jira-importer** [comment]
**[Ron Castel](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rcastel)** commented

This issue is effectively fixed since Maven 3.5.2, by permitting to set the value used during child's pom inheritance assembly of the scm.url property, among several others. See:

http://maven.apache.org/ref/3.5.2/maven-model-builder/index.html#Inheritance_Assembly

I tested it and it works nicely in my projects.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

As described: Works as designed and a known solution is available.
